src/cargo/ops/cargo_compile.rs

```diff
@@ -289,12 +293,53 @@ pub fn compile_ws<'a>(
         unit_graph::emit_serialized_unit_graph(&bcx.roots, &bcx.unit_graph)?;
         return Compilation::new(&bcx);
     }
-
     let _p = profile::start("compiling");
     let cx = Context::new(&bcx)?;
     cx.compile(exec)
 }
 
+pub fn print<'a>(
+    ws: &Workspace<'a>,
+    options: &CompileOptions,
+    print_opt_value: &str,
+) -> CargoResult<()> {
+    let CompileOptions {
+        ref build_config,
+        ref target_rustc_args,
+        ..
+    } = *options;
+    let config = ws.config();
+    let rustc = config.load_global_rustc(Some(ws))?;
+    for (index, kind) in build_config.requested_kinds.iter().enumerate() {
+        if index != 0 {
+            drop_println!(config);
+        }
+        let rustflags = env_args(
+            config,
+            &build_config.requested_kinds,
+            &rustc.host,
+            None,
+            *kind,
+            "RUSTFLAGS",
+        )?;
+        let mut process = rustc.process();
+        process.args(&rustflags);
+        if let Some(args) = target_rustc_args {
+            process.args(args);
+        }
+        if let CompileKind::Target(t) = kind {
+            process.arg("--target").arg(t.short_name());
+        }
+        process.arg("--print").arg(print_opt_value);
+        let output = process.exec_with_output()?;
```

Comment:
Instead of capturing the output here I think it's best to jsutt forward the output to Cargo's own stdout/stderr streams

Reply:
I am sorry, but I am not sure how to forward the output to Cargo's own stdout/stderr streams. Do you have an example of doing this within the Cargo codebase for another process/cmd? I looked into the `ProcessBuilder` type and initially tried the `exec` method, but it just hanged. I believe the equivalent method in the `std::process::Command` type would be `status()` because it says it the child (rustc) inherients stdio from the parent (Cargo). Would this be the same as "forwarding"?

Reply:
I believe you can basically call `exec` instead of `exec_with_output` here